src/js/map/maps.js

```diff
@@ -98,26 +100,48 @@ export class MapControl extends Observable {
         this.myEventForwarder.enable();
     }
 
+    async loadSubindicatorData(geo) {
+        this.triggerEvent("map.choropleth.loading");
+
+        let response;
+        let savedObj = this.choroplethData.filter((x) => {
```

Comment:
why don't you use an object instead of an array?

then you could do 
```js
let savedObj = this.choroplethData[geo];
```

and then you could do (have not tried it out but should work:

```js
let savedObj = this.choroplethData[geo] || getChoroplethData(...);

async getChoroplethData(geo) {
this.choroplethData[geo] = await this.api.loadChoroplethData(this.config.profile, geo, 1);
return this.choroplethData[geo];
}

Reply:
I used an array instead of an object because at the time I coded this part there was one thing I left for later : `indicatorId`. So we actually have 2 parameters to filter with.

This is how this function looks now : 
```
    async loadSubindicatorData(geo, indicatorId) {
        this.triggerEvent("map.choropleth.loading");

        let response;
        let savedObj = this.choroplethData.filter((x) => {
            return x.geography === geo && x.indicatorId == indicatorId
        })[0];

        if (savedObj !== null && typeof savedObj !== 'undefined') {
            response = savedObj.data;
        } else {
            response = await this.api.loadChoroplethData(this.config.profile, geo, indicatorId);
            this.choroplethData.push({
                'geography': geo,
                'data': response,
                'indicatorId': indicatorId
            });
        }

        return response;
    }
```